ExpressionUtils/Evaluating/ExpressionInterpreter.cs

```diff
@@ -118,6 +118,8 @@ class ExpressionInterpretationVisitor : ExpressionResultVisitor<object> {
 							return left << right;
 						case ExpressionType.RightShift:
 							return left >> right;
+                        case ExpressionType.ArrayIndex:
+                            return ((Array) left).GetValue(right);

```

Comment:
Since left is `dynamic`, is that cast actually necessary?

Reply:
Alright I removed the cast